src/area/uploadArea.ts

```diff
@@ -362,10 +370,21 @@ class UploadArea {
 
     private isFileSizeValid(file: File): boolean {
         var maxFileSize = this.options.maxFileSize * 1024 * 1024; // max file size in bytes
-        if (file.size > maxFileSize) return false;
+        if (file.size > maxFileSize || file.size === 0) return false;
         return true;
     }
 
+    private isFileTypeInvalid(file: File): boolean {
+        if (file.name && (this.options.accept.trim() !== '*' || this.options.accept.trim() !== '*.*') && this.options.validateExtension) {
```

Comment:
It should be able to detect whether there is extenstion list or MIME type definition.